npm/test/upload.test.ts

```diff
@@ -0,0 +1,111 @@
+import { ClientFileProcessing } from "../src/clientfileprocessing"
+import { KeycloakInstance } from "keycloak-js"
+import { TdrFile } from "@nationalarchives/file-information"
+import { ClientFileMetadataUpload } from "../src/clientfilemetadataupload"
+import { GraphqlClient } from "../src/graphql"
+import { UploadFiles } from "../src/upload"
+
+jest.mock("../src/clientfilemetadataupload")
+
+beforeEach(() => jest.resetModules())
+
+const mockKeycloak: KeycloakInstance<"native"> = {
+  init: jest.fn(),
+  login: jest.fn(),
+  logout: jest.fn(),
+  register: jest.fn(),
+  accountManagement: jest.fn(),
+  createLoginUrl: jest.fn(),
+  createLogoutUrl: jest.fn(),
+  createRegisterUrl: jest.fn(),
+  createAccountUrl: jest.fn(),
+  isTokenExpired: jest.fn(),
+  updateToken: jest.fn(),
+  clearToken: jest.fn(),
+  hasRealmRole: jest.fn(),
+  hasResourceRole: jest.fn(),
+  loadUserInfo: jest.fn(),
+  loadUserProfile: jest.fn(),
+  token: "fake-auth-token"
+}
+
+class ClientFileMetadataUploadSuccess {
+  upload: (consignmentId: string, files: TdrFile[]) => Promise<void> = async (
+    consignmentId: string,
+    files: TdrFile[]
+  ) => {
+    return Promise.resolve()
+  }
+}
+
+class ClientFileMetadataUploadFailure {
+  upload: (consignmentId: string, files: TdrFile[]) => Promise<void> = async (
+    consignmentId: string,
+    files: TdrFile[]
+  ) => {
+    return Promise.reject(Error("Client file metadata upload failed"))
+  }
+}
+
+const mockSuccess: () => void = () => {
+  const mock = ClientFileProcessing as jest.Mock
+  mock.mockImplementation(() => {
+    return new ClientFileMetadataUploadSuccess()
+  })
+}
+
+const mockFailure: () => void = () => {
+  const mock = ClientFileProcessing as jest.Mock
+  mock.mockImplementation(() => {
+    return new ClientFileMetadataUploadFailure()
+  })
+}
+
+function setUpUploadFiles(): UploadFiles {
+  const client = new GraphqlClient("test", mockKeycloak)
+  const clientFileMetadataUpload: ClientFileMetadataUpload = new ClientFileMetadataUpload(
+    client
+  )
```

Comment:
This function and the two above it aren't being used.

Reply:
Removed